### PR TITLE
os/board/rtl8730e: Add wakeup reason for PM

### PR DESCRIPTION
**Pull request by edwakuwaku**
- Add wakeup reason for PM
- Apply macro to filter redundant global variables

### CODE REVIEW

File: os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c [hunk 1]
```diff
@@ -240,6 +240,8 @@ void tizenrt_pre_sleep_processing(uint32_t *expected_idle_time)
 	/* ms =x*1000/32768 = (x *1000) >>15 */
 	ms_passed = (u32)((((u64)tick_passed) * 1000) >> 15);
 
+	u16 wakeup_reason = HAL_READ16(SYSTEM_CTRL_BASE_LP, REG_LSYS_DUMMY_090);
```
Comment (edwakuwaku) [inline os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c:243]: Hi @gSahitya-samsung , this wakeup reason has followed the enum in your PR. But as your PR is not merged, I couldn't feed this directly into the pm_wakehandler. In the meantime, please use it as a patch for your test first.
Thanks.

Comment (gSahitya-samsung) [reply os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c:243]: Verified it for LOG_UART interrupts

**Comment by gSahitya-samsung** [comment]
Verified PM Metrics wakeup sources counts with this PR

```
pm_print_metrics:  WAKEUP SOURCES |      COUNTS      
pm_print_metrics: ----------------|------------------
pm_print_metrics:         UNKNOWN |      0 (  0.00%) 
pm_print_metrics:             BLE |      0 (  0.00%) 
pm_print_metrics:            WIFI |      0 (  0.00%) 
pm_print_metrics:    UART CONSOLE |     14 ( 93.33%) 
pm_print_metrics:      UART TTYS2 |      0 (  0.00%) 
pm_print_metrics:            GPIO |      0 (  0.00%) 
pm_print_metrics:        HW TIMER |      1 (  6.67%)
```

**Review by ewoodev — APPROVED** [review]

File: os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c [hunk 2]
```diff
@@ -240,12 +243,14 @@ void tizenrt_pre_sleep_processing(uint32_t *expected_idle_time)
...
+
 #ifndef CONFIG_PLATFORM_TIZENRT_OS
 	vTaskStepTick(ms_passed); /*  update kernel tick */
 #else
 #ifdef CONFIG_PM_TICKSUPPRESS
 	if (tizenrt_sleep_handler) {
-		tizenrt_sleep_handler((u64)ms_passed); /*  update kernel tick */
+		tizenrt_sleep_handler((clock_t)ms_passed, (pm_wakeup_reason_code_t)wakeup_reason); /*  update kernel tick */
```
Comment (ewoodev) [inline os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c:253]: void up_pm_board_sleep(void (*handler)(clock_t, pm_wakeup_reason_code_t))

Could you please change to use this handler?
we will remove up_register_wakehandler()
I added up_register_wakehandler temporary.

File: os/pm/pm_initialize.c [hunk 3]
```diff
@@ -113,11 +113,6 @@ void pm_initialize(void)
 	g_pmglobals.stime = clock_systimer();
 	g_pmglobals.btime = clock_systimer();
 
-#ifdef CONFIG_PM_TICKSUPPRESS
-	/* Registers a handler to be called when the core wakes up */
```
Comment (gSahitya-samsung) [inline os/pm/pm_initialize.c:117]: Please separate pm_initialize.c and arch.h changes into different commit

Comment (edwakuwaku) [reply os/pm/pm_initialize.c:117]: done~

File: os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c (hunk 2 above)
Comment (edwakuwaku) [reply os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c]: done~

**Review by gSahitya-samsung — APPROVED** [review]
Verified with PM Metrics
```
pm_print_metrics:  WAKEUP SOURCES |      COUNTS      
pm_print_metrics: ----------------|------------------
pm_print_metrics:         UNKNOWN |      0 (  0.00%) 
pm_print_metrics:             BLE |      0 (  0.00%) 
pm_print_metrics:            WIFI |      0 (  0.00%) 
pm_print_metrics:    UART CONSOLE |     14 ( 93.33%) 
pm_print_metrics:      UART TTYS2 |      0 (  0.00%) 
pm_print_metrics:            GPIO |      0 (  0.00%) 
pm_print_metrics:        HW TIMER |      1 (  6.67%) 
```

**Review by ewoodev — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**